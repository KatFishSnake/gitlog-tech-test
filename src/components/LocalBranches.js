import { useState } from "react";
import styled from "styled-components";

import useFetch from "../hooks/useFetch";
import { mockLocal } from "../mockData";
import NavBranch from "./NavBranch";

export const StyledNavHeader = styled.h4`
  color: #81858b;
`;

const LocalBranches = () => {
  const [active, setActive] = useState(null);
  const { loading, data: localBranches } = useFetch(mockLocal);

  return (
    <div className="flex flex-col mb-4">
      <StyledNavHeader className="mb-4 text-xxs not-italic font-bold leading-3 tracking-normal text-left">
        LOCAL BRANCHES
      </StyledNavHeader>
      <div className="text-xs">
        {loading
          ? "Loading..."
          : localBranches.map((item) => (
              <NavBranch
                key={item.id}
                label={item.label}
                active={item.id === active}
                onClick={() => {
                  setActive(item.id);
                }}
              />
            ))}
      </div>
    </div>
  );
};

export default LocalBranches;
