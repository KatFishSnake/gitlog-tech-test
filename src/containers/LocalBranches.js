import { useState } from "react";

import useFetch from "../hooks/useFetch";
import { mockLocal } from "../config";
import NavItem from "../components/NavItem";
import { StyledNavHeader } from "../components/styled";
import Loader from "../components/Loader";

const LocalBranches = () => {
  const [active, setActive] = useState(null);
  const { loading, data: localBranches } = useFetch(mockLocal);

  return (
    <div className="flex flex-col mb-4">
      <StyledNavHeader className="mb-4 text-xxs not-italic font-bold leading-3 tracking-normal text-left">
        LOCAL BRANCHES
      </StyledNavHeader>
      <div className="text-xs">
        {loading ? (
          <Loader />
        ) : (
          localBranches.map((item) => (
            <NavItem
              key={item.id}
              label={item.label}
              active={item.id === active}
              onClick={() => {
                setActive(item.id);
              }}
            />
          ))
        )}
      </div>
    </div>
  );
};

export default LocalBranches;
