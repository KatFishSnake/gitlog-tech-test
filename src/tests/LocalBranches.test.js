import { render } from "@testing-library/react";

import LocalBranches from "../containers/LocalBranches";

test("Renders LocalBranches consistently", () => {
  const { container } = render(<LocalBranches />);
  expect(container).toMatchSnapshot();
});
