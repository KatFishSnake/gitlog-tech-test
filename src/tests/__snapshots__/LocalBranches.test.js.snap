// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders LocalBranches consistently 1`] = `
<div>
  <div
    class="flex flex-col mb-4"
  >
    <h4
      class="sc-bdfBwQ iHCwhp mb-4 text-xxs not-italic font-bold leading-3 tracking-normal text-left"
    >
      LOCAL BRANCHES
    </h4>
    <div
      class="text-xs"
    >
      <span>
        Loading...
      </span>
    </div>
  </div>
</div>
`;
